src/main/java/base/gameobjects/animals/Pig.java

```diff
@@ -0,0 +1,53 @@
+package base.gameobjects.animals;
+
+import static base.Game.TILE_SIZE;
+
+import base.Game;
+import base.gameobjects.Animal;
+import base.graphicsservice.Rectangle;
+import base.graphicsservice.Sprite;
+import base.map.GameMap;
+import base.navigationservice.Direction;
+
+public class Pig extends Animal{
+
+	public Pig(Sprite playerSprite, int startX, int startY, int speed) {
+		super(playerSprite, startX, startY, speed);
+	}
+
+	@Override
+    public boolean handleMouseClick(Rectangle mouseRectangle, Rectangle camera, int xZoom, int yZoom) {
+        logger.info("Click on Animal: rat");
+        return false;
+    }
+	
+	protected void handleMoving(GameMap gameMap, Direction direction) {
+        if (unwalkableInThisDirection(gameMap, direction)) {
```

Comment:
I would suggest to change initial image and re-arrange pigs on it. So first row will be pig facing forward, second row - pig facing left, third row - pig facing up, last row - pig facing right.
Otherwise you will have bugs when pig is meeting obstacles.